src/directory_sync/operations/list_directories.rs

```diff
@@ -0,0 +1,206 @@
+use async_trait::async_trait;
+use reqwest::StatusCode;
+use serde::Serialize;
+
+use crate::directory_sync::{Directory, DirectorySync, DirectoryType};
+use crate::organizations::OrganizationId;
+use crate::{KnownOrUnknown, PaginatedList, PaginationOptions, WorkOsError, WorkOsResult};
+
+/// The options for [`ListDirectories`].
+#[derive(Debug, Serialize)]
+pub struct ListDirectoriesOptions<'a> {
+    /// The domain of a directory
+    pub domain: Option<&'a String>,
+
+    /// Searchable text to match against Directory names.
+    pub search: Option<&'a String>,
+
+    /// The pagination options to use when listing directories.
+    #[serde(flatten)]
+    pub pagination: PaginationOptions<'a>,
+
+    /// The ID of the organization to list directories for.
+    pub organization_id: Option<&'a OrganizationId>,
+
+    /// The type of directories to list.
+    #[serde(rename = "directory_type")]
+    pub r#type: &'a Option<KnownOrUnknown<DirectoryType, String>>,
+}
+
+impl<'a> Default for ListDirectoriesOptions<'a> {
+    fn default() -> Self {
+        Self {
+            pagination: PaginationOptions::default(),
+            organization_id: None,
+            r#type: &None,
+            domain: None,
+            search: None,
+        }
+    }
+}
+
+/// [WorkOS Docs: List Directories](https://workos.com/docs/reference/directory-sync/directory/list)
+#[async_trait]
+pub trait ListDirectories {
+    /// Retrieves a list of [`Directories`]s.
```

Comment:
This should be singular since we're referring to the `Directory` type:

```suggestion
    /// Retrieves a list of [`Directory`]s.
```